core-common/src/main/java/org/zalando/nakadi/enrichment/MetadataEnrichmentStrategy.java

```diff
@@ -39,12 +40,30 @@ public void enrich(final BatchItem batchItem, final EventType eventType) throws
         }
     }
 
+    @Override
+    public void enrich(final NakadiRecord nakadiRecord, final EventType eventType) throws EnrichmentException {
+        final var metadata = nakadiRecord.getMetadata();
+        metadata.setPublishedBy(getPublisher());
+        final DateTime dateTime = new DateTime(DateTimeZone.UTC);
+        metadata.setReceivedAt(dateTime.toString());
+        metadata.setEventType(eventType.getName());
+        if ("".equals(metadata.getFlowId())) {
+            metadata.setFlowId(FlowIdUtils.peek());
+        }
+        metadata.setPartition(nakadiRecord.getPartition());
+        metadata.setSchemaVersion(eventType.getSchema().getVersion().toString());
```

Comment:
I'd expect the event type schema version to be provided by the client, and not enriched in case of binary record.

Reply:
Good point, I'll fix this.